### ISSUE TITLE
Save All and Save All Levels commands don't always create level backups

### ISSUE THREAD

**Issue by artisteacher** [opentoonz/opentoonz#3091]
### Issue Summary

I recently opened an old project in 1.4 (created before level backups were a thing). The scene has several raster levels (tifs). I haven't made changes to the drawings themselves, but have changed how they are utilized in the scene. "Saving All" and "Save All Levels" does not create backup levels. If I individually save the levels themselves with "Save Level" a backup is created.
### Steps to Reproduce

1. Open a scene that doesn't have any level backups (level type does not matter)
2. Use "Save All" or "Save All Levels"

### Expected Results

If there are no existing level backups, the backup should be created - even if the level is not "dirty".
### Actual Results

No level backups are created, unless the drawings themselves are changed.
### System Information
* **OpenToonz Version**: 1.4
* **Operating System**: macOS 10.14.6
* **CPU**: 
* **Memory**: 
* **Graphics Card**: 
* **Graphics Tablet**:

**Comment by gab3d** [comment]
> I haven't made changes to the drawings themselves, but have changed how they are utilized in the scene.

I wonder, but the changes you've made to the scene ("how they are utilized") are being saved, right?

**Comment by artisteacher** [comment]
Yes, scene changes are saving & the backup file for the scene is generated.

**Comment by gab3d** [comment]
Great! 
So, what I cannot make clear is what the utility of backing-up the unchanged levels in this case is. 💁‍♂️
I mean, if the levels have not undergone any changes in the OT session, and the scene is correctly saved, there should be no additional benefit in having a duplication of the unchanged levels, am I right? 
Surely I must be missing something in your workflow... 🤔

**Comment by artisteacher** [comment]
My students & I have had issues with levels going completely missing #2581 - they show up in the Xsheet with red numbers but they aren't anywhere to be found on the computer. It's inconsistent & hard to track since the problem must happen at the end of the previous session using OT. If there is a backup of the levels, we can load them to replace the missing levels. Otherwise we have to start from scratch.

Additionally, using the "Save Level" command on a single level does generate a backup - even if there were not any changes made to the drawings. I think "Save All" and "Save All Levels" should be consistent with that. This is assuming of course that backups are enabled in preferences.

**Comment by gab3d** [comment]
> My students & I have had issues with levels going completely missing (...)

I don't know how is your class hardware set-up, but from my own experience, there are two main reasons why students lose their saved files: 
1. They save them on a frozen drive, so when the computer is switched back on, the files are gone.
2. They save it on a public drive, and sometimes other students delete those folders, so they lost their jobs for good.
3. They created the levels in OT, but never saved them (they usually use the Save Scene command, believing that they've just saved everything), so when they open the scene again they find their levels are lost.

> If there is a backup of the levels, we can load them to replace the missing levels.

In any of those 3 cases above, I guess a backup won't help you much either, but perhaps neither of those cases are applicable to your set-up...?

More importantly, what you assume is that OT deletes the files. So, does it happen with any type of levels or that can just happen to any level type? 🤔

**Comment by manongjohn** [comment]
> More importantly, what you assume is that OT deletes the files. 

Sadly, it has been proven that OT deletes files accidentally due to certain bugs so yes, you can assume OT is doing this.

In some save operations, what happens before something is saved is the original file is deleted, a temp file is created and then renamed to replace the original file.  Sometimes there is an issue with the temp file creation, or even the rename,  but by then the file has been deleted and can't be recovered.  Backup should be occurring before that happens, but if it somehow fails to create the backup and still manages to delete the original file, that is also an issue.

As for backing up:
Scenes save and backup should always occurs whenever an appropriate save action is initiated (Save All, Save Scene).  Doesn't matter if there was a change to the scene or not.

Levels only save if it has detected that the level changed (is flagged as "dirty").  Unfortunately, we do have issues where a level is not being flagged appropriately so those changes may be lost if there is no further action that causes the level to become "dirty".  

Need to review how things are being saved, again, and probably put in more safeguards to prevent total loss.

**Comment by artisteacher** [comment]
> there are two main reasons why students lose their saved files:
 
1 & 2: We're saving locally, to individual student accounts (User/Documents/OpenToonz)
3: Auto-save is set to run every 7 minutes & I have students saving manually with cmd+S (Save All)

While user error is always possible, as @manongjohn said, OT definitely deletes the files in some cases.
